btf/marshal_test.go

```diff
@@ -86,19 +77,9 @@ func BenchmarkBuildVmlinux(b *testing.B) {
 	b.ReportAllocs()
 	b.ResetTimer()
 
-	types := spec.types
-	strings := spec.strings
-
 	for i := 0; i < b.N; i++ {
-		enc := newEncoder(encoderOptions{ByteOrder: internal.NativeEndian}, newStringTableBuilderFromTable(strings))
-
-		for _, typ := range types {
-			if _, err := enc.Add(typ); err != nil {
-				b.Fatal(err)
-			}
-		}
-
-		_, err := enc.Encode()
+		var buf bytes.Buffer
```

Comment:
Is there something to be gained by declaring `buf` outside the loop and resetting here?

Reply:
Yeah it would remove allocations for buf I think, but I figured that would mean we don't measure what it costs to build a single vmlinux.